### DISCUSSION TITLE
How does the cswinrt.exe tool know which "C# version" to use to generate the .cs files ?

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/CsWinRT#1145] [Q&A]
the cswinrt.exe tool generates `*.cs` files from .winmd files.  
I don't see any switch to specify the "C# version" in the cswinrt.exe tool !

How does the cswinrt.exe tool know which "C# version" to use to generate the `*.cs` files  from .winmd files? 

end goal : is to understand how "language versions" fit in the WinRT language projections.

--
GitHub Ghost

**Comment by manodasanW** [comment]
Today, the NET 5/6/7 sources generated by CsWinRT are compatible with C# 9.0 and isn't configurable independently.  We haven't started using features that are available only in later C# versions, but we may choose to in the future.  When we do, we will possibly decide based on the TFM for which version that the sources are generated for.  But that will be something that we decide when we get to using those new language features.  Is there something particular that you were concerned about?

**Reply by ghost** [reply]
> But that will be something that we decide when we get to using those new language features.

(just like C++/WinRT v3 targeting C++23 as min version.). Using newer C# language syntaxes will require a new C#/WinRT version maybe   C#/WinRT v2.   Is that right ?
